msal/application.py

```diff
@@ -825,8 +819,7 @@ def acquire_token_by_username_password(
             - A successful response would contain "access_token" key,
             - an error response would contain "error" and usually "error_description".
         """
-        if not self.client:
-            self._setup_before_first_network_call()
+        self._get_client()
```

Comment:
This line functionally works. It is a different pattern here, when comparing to your implementations in other `acquire_token_with_xyz()` methods. Which one do you think is better, and why? :-)

Reply:
So, for this one we need some attributes of Authority even before we need our client. In previous implementation, those would get configured in the initialization of the authority in the app __init__ so we can do network calls in authority and client initialization as _get_client().obtain_token_xyz(). For this one, since we need the authority attributes, I chose to initialize the client in the first step.

Reply:
Oh I see. Then I would suggest, instead of adding this line here, we add an explicit `self.authority.initialize()` at line 828.5 and then go back to the same `return self._get_client().obtain_tokan_by_xyz(...)` pattern we used elsewhere.